### PR TITLE
test/librbd: fix group_info.name leaks in TestGroup.add_image

### PR DESCRIPTION
**Pull request by idryomov**
Introduced in commit c04e8c3374e0 ("test/librbd: add get_group test").

Signed-off-by: Ilya Dryomov <idryomov@gmail.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/test/librbd/test_Groups.cc [hunk 1]
```diff
@@ -99,8 +99,9 @@ TEST_F(TestGroup, add_image)
 
   rbd_group_info_t group_info;
   ASSERT_EQ(0, rbd_get_group(image, &group_info, sizeof(group_info)));
-  ASSERT_EQ(0,strcmp("", group_info.name));
+  ASSERT_EQ(0, strcmp("", group_info.name));
   ASSERT_EQ(RBD_GROUP_INVALID_POOL, group_info.pool);
+  free(group_info.name);
```
Comment (trociny) [inline src/test/librbd/test_Groups.cc:104]: nit: use `rbd_group_info_cleanup` instead?

**Review by trociny — APPROVED** [review]
Otherwise LGTM

File: src/test/librbd/test_Groups.cc (hunk 1 above)
Comment (idryomov) [reply src/test/librbd/test_Groups.cc:104]: I wonder how I managed to miss that function...

**Comment by idryomov** [comment]
jenkins retest this please

**Comment by idryomov** [comment]
No related failures:

https://pulpito.ceph.com/dis-2022-01-08_10:51:32-rbd-wip-dis-testing-distro-default-smithi/
https://pulpito.ceph.com/dis-2022-01-10_22:21:45-rbd-wip-dis-testing-distro-default-smithi/

**Pull request merged**